flake8_type_checking/checker.py

```diff
@@ -566,6 +576,10 @@ def visit_If(self, node: ast.If) -> Any:
                 # first element in the else block - 1
                 start_of_else_block = node.orelse[0].lineno - 1
 
+            # mark all the top-level statements
+            for stmt in node.body:
+                setattr(stmt, TOP_LEVEL_PROPERTY, True)
```

Comment:
How about we name it `TOP_LEVEL_IF_PROPERTY`?

Reply:
That reads to me like the `if` is top level rather than the statement. I tried to come up with a more descriptive name, but it ends up being too long. Most unambiguous would be something like `TYPE_CHECKING_TOP_LEVEL_STMT_PROPERTY`. If you don't mind it being that long-winded I'm happy to change it.

Reply:
I realize now that there are a bunch of corner-cases still here, it's quite common to have an `if TYPE_CHECKING` block inside a class body, so declarations should remain scoped to the class body. It's probably more robust if we mark all the statements in global scope with `GLOBAL_STMT_PROPERTY`. This should be fairly straightforward, although we would need to add some recursion for control flow in global scope, since control flow does not open a new scope, while function/class definitions do.

Then we could combine `GLOBAL_STMT_PROPERTY` and `in_type_checking_block` to accomplish the same check or add a second marker, like we were doing here, but only if the `if` is also marked global.

Reply:
I ended up changing the implementation accordingly and added an extra statement to one of the inverse regression tests to ensure this works.

We may want to consider ignoring imports that aren't in global scope in the same way, although less common it is possible that there's an import inside a function definition to resolve a cyclic dependency, which should probably not count towards the imports we check.